hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/multimap/MultiMapKeySetMessageTask.java

```diff
@@ -54,7 +54,7 @@ protected OperationFactory createOperationFactory() {
 
     @Override
     protected Object reduce(Map<Integer, Object> map) {
-        List<Data> keys = new ArrayList<Data>();
+        Set<Data> keys = new HashSet<Data>();
```

Comment:
How do we know no duplicates will exist? If there are duplicates, this change may cause loss of data.